src/hkdf/hkdf.ts

```diff
@@ -0,0 +1,40 @@
+import { ProviderCrypto } from "../provider";
+import { KeyUsages } from "../types";
+import { BufferSourceConverter } from "../utils";
+
+export abstract class HkdfProvider extends ProviderCrypto {
+
+  public name = "HKDF";
+  public hashAlgorithms = ["SHA-1", "SHA-256", "SHA-384", "SHA-512"];
+  public usages: KeyUsages = ["deriveKey", "deriveBits"];
+
+  public checkAlgorithmParams(algorithm: HkdfParams) {
+    // hash
+    this.checkRequiredProperty(algorithm, "hash");
+    this.checkHashAlgorithm(algorithm.hash as Algorithm, this.hashAlgorithms);
+
+    // salt
+    this.checkRequiredProperty(algorithm, "salt");
+    if (!BufferSourceConverter.isBufferSource(algorithm.salt)) {
+      throw new TypeError("salt: Is not of type '(ArrayBuffer or ArrayBufferView)'");
+    }
+
+    // info
+    this.checkRequiredProperty(algorithm, "info");
```

Comment:
'info' is optional for HKDF

Reply:
https://www.w3.org/TR/WebCryptoAPI/#dfn-HkdfParams